### PR TITLE
[Merged by Bors] - feat(Geometry/Manifold/VectorBundle/Tangent): tangentCoordChange

### PR DESCRIPTION
**Pull request by winstonyin**
We define `tangentCoordChange` as a convenient abbreviation for coordinate changes on the tangent bundle. We also restate the axioms of `VectorBundleCore` as lemmas involving `extChartAt`.

Currently, we need to write `(tangentBundleCore I M).coordChange (achart H x) (achart H y)`, referring explicitly to the atlas of `M`. Since `tangentBundleCore` uses the same base sets as the preferred charts of the base manifold, we wish to work directly with points `x y : M` and the preferred extended charts at those points (`extChartAt`).

We find this definition and related lemmas useful in #8483 in shortening proofs.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — APPROVED** [review]
The added lemmas look plausible and straightforward; their uses in the dependent PR make this compelling to me. (Please address the style comment.)

File: Mathlib/Geometry/Manifold/VectorBundle/Tangent.lean [hunk 1]
```diff
@@ -124,6 +124,47 @@ theorem tangentBundleCore_coordChange_achart (x x' z : M) :
   rfl
 #align tangent_bundle_core_coord_change_achart tangentBundleCore_coordChange_achart
 
+section tangentCoordChange
+
+/-- Since `tangentBundleCore` has the same base sets as the preferred charts of the base manifold,
+  we define `tangentCoordChange I x y` as a convenient abbreviation for a change of coordinates
+  between the preferred charts at `x y : M`. -/
+abbrev tangentCoordChange (x y : M) := (tangentBundleCore I M).coordChange (achart H x) (achart H y)
```
Comment (ocfnash) [inline Mathlib/Geometry/Manifold/VectorBundle/Tangent.lean:132]: Let's help the reader and give the type here:
```suggestion
abbrev tangentCoordChange (x y : M) : M → E →L[𝕜] E :=
  (tangentBundleCore I M).coordChange (achart H x) (achart H y)
```

File: Mathlib/Geometry/Manifold/VectorBundle/Tangent.lean [hunk 2]
```diff
@@ -124,6 +124,47 @@ theorem tangentBundleCore_coordChange_achart (x x' z : M) :
   rfl
 #align tangent_bundle_core_coord_change_achart tangentBundleCore_coordChange_achart
 
+section tangentCoordChange
+
+/-- Since `tangentBundleCore` has the same base sets as the preferred charts of the base manifold,
+  we define `tangentCoordChange I x y` as a convenient abbreviation for a change of coordinates
+  between the preferred charts at `x y : M`. -/
```
Comment (ocfnash) [inline Mathlib/Geometry/Manifold/VectorBundle/Tangent.lean:131]: I think I'd reverse the order of emphasis here and start with the fact that `x` and `y` are really indexing charts. Maybe something like:
```suggestion
/-- In a manifold `M`, given two preferred charts indexed by `x y : M`, `tangentCoordChange I x y` is the family of derivatives of the corresponding change-of-coordinates map. It takes junk values outside the intersection of the sources of the two charts.

Note that this definition takes advantage of the fact that `tangentBundleCore` has the same base sets as the preferred charts of the base manifold. -/
```

**Review by ocfnash — APPROVED** [review]
Thanks!

bors d+

**Pull request closed without merging**